From: Felix Krull <f_krull@gmx.de>
Date: Sun, 3 Jul 2016 14:45:11 +0200
Subject: libffi-shared

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 088ab00..55c2e9e 100644
--- a/setup.py
+++ b/setup.py
@@ -2008,7 +2008,7 @@ class PyBuildExt(build_ext):
                           'ffi_wrapper_h'.format(ffi_h))
         ffi_lib = None
         if ffi_inc is not None:
-            for lib_name in ('ffi_convenience', 'ffi_pic', 'ffi'):
+            for lib_name in ('ffi', 'ffi_convenience', 'ffi_pic', 'ffi'):
                 if (self.compiler.find_library_file(lib_dirs, lib_name)):
                     ffi_lib = lib_name
                     break
