# DP: Ignore errors in the profile task.

Index: b/Makefile.pre.in
===================================================================
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -494,7 +494,16 @@ build_all_generate_profile:
 
 run_profile_task:
 	: # FIXME: can't run for a cross build
-	$(RUNSHARED) ./$(BUILDPYTHON) $(PROFILE_TASK)
+	task="$(PROFILE_TASK)"; \
+	case "$$task" in \
+	*-s\ *) \
+	  $(RUNSHARED) ./$(BUILDPYTHON) $$task; \
+	  while [ -f $(srcdir)/build/pynexttest ]; do \
+	    $(RUNSHARED) ./$(BUILDPYTHON) $$task; \
+	  done;; \
+	*) \
+	  $(RUNSHARED) ./$(BUILDPYTHON) $$task; \
+	esac
 
 build_all_use_profile:
 	$(MAKE) all CFLAGS_NODIST="$(CFLAGS) -fprofile-use -fprofile-correction"
